### PR TITLE
Use the Flickr API for MozCamp

### PR DESCRIPTION
**Pull request by xabolcs**
Flickr needs an `api_key`. That `api_key` needs a Yahoo account.

I updated the MozCamp extension to use the required Flicker API.

Please review!

### CODE REVIEW

**Comment by xabolcs** [comment]
If these changes are OK, I happily update the Flicker extension too.

File: MozCamp/bootstrap.js [hunk 1]
```diff
@@ -20,11 +20,11 @@ function startup(aData, aReason) {
...
 		if(xhr.status == 200) {
 			try {
-				let m = JSON.parse(xhr.responseText);
+				let m = JSON.parse(xhr.responseText).photos.photo;
 				m = m.splice(0, m.length * 40 / 100);
 				m.sort(function() 0.5-Math.random());
-				m = m.shift().sizes;
-				var URL = (m.h || m.l || m.o).url;
+				m = m.shift();
+				var URL = m.url_h || m.url_o;
```
Comment (xabolcs) [inline MozCamp/bootstrap.js:27]: I didn't find anything in the API for the `m.l`.

File: MozCamp/bootstrap.js [hunk 2]
```diff
@@ -73,9 +73,22 @@ function startup(aData, aReason) {
 		}
 	},false);
 	
-	let date = new Date(Date.now() - (4 * 30 * 24 * 60 * 60 * 1000)).toISOString().split('T').shift().replace('-','','g');
-	xhr.open('GET', 'http://www.flickr.com/search?data=1&q=mozilla%20summit%20OR%20mozcamp&s=int&mt=photos&d=taken-'+date+'-&append=1', true);
-	xhr.send();
+	let date = new Date(Date.now() - (4 * 30 * 24 * 60 * 60 * 1000));
+	let formData = new flickerSearchForm();
+	formData.append("extras", "url_o,url_h");
+	formData.append("tags", "mozsummit,mozcamp");
```
Comment (xabolcs) [inline MozCamp/bootstrap.js:79]: With free text I wasn't able to search as the old code did. So I switched to tag based search from free text based search. The new result may differ from the old result set.

File: MozCamp/bootstrap.js [hunk 3]
```diff
@@ -73,9 +73,22 @@ function startup(aData, aReason) {
...
+	formData.append("min_taken_date", (date.getTime() / 1000));
+	xhr.open('POST', 'https://api.flickr.com/services/rest/', true);
+	xhr.send(formData);
+}
+
+function flickerSearchForm () {
+	let formData = Cc['@mozilla.org/files/formdata;1'].createInstance(Ci.nsIDOMFormData);
+	formData.append("method", "flickr.photos.search");
+	formData.append("api_key", "6319252bea53c42693c74970e245542e");
```
Comment (xabolcs) [inline MozCamp/bootstrap.js:88]: `api_key` is required, so I generated one, a non-commercial. Feel free to create your own.

**Comment by xabolcs** [comment]
@diegocr, any feedback here?

**Comment by diegocr** [comment]
Hi @xabolcs, sorry for the delay.

I reviewed your changes and tested the extension, all looks good. Thank you!

Just one thing, what are the limitations of the free api key? i suppose it has maximum number of requests or something like that. Anyway, i'll just merge this since the previous method is not returning anything.

**Comment by xabolcs** [comment]
The [limits](https://www.flickr.com/services/developer/api/) are 3600 queries per hour.

> If your application stays under 3600 queries per hour across the whole key (which means the aggregate of all the users of your integration), you'll be fine.

**Comment by xabolcs** [comment]
Thanks for the merge! :+1:

**Comment by xabolcs** [comment]
@diegocr, may I provide an update to the Flicker extension too?

**Comment by diegocr** [comment]
Sure, feel free to update that one too.

Thank you for the info on the limits, and everything else :)

File: MozCamp/bootstrap.js (hunk 1 above)
Comment (xabolcs) [inline MozCamp/bootstrap.js:27]: `url_l` is the new `m.l`.

**Pull request merged**